### PR TITLE
Improve generator

### PR DESCRIPTION
**Pull request by rustamwin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️/❌
| New feature?  | ✔️/❌
| Breaks BC?    | ✔️/❌
| Tests pass?   | ✔️/❌
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

TODO:
- [x] validation

### CODE REVIEW

File: src/Generator/AbstractGenerator.php [hunk 1]
```diff
@@ -325,6 +337,76 @@ public function render(string $template, array $params = []): string
...
+     * @param DataSetInterface $dataSet
+     * @return Result
+     */
+    public function validateTemplate($value, DataSetInterface $dataSet): Result
+    {
+        $result = new Result();
+        $templates = $dataSet->getTemplates();
+        if ($templates === []) {
+            return $result;
+        } elseif (!isset($templates[$value])) {
```
Comment (xepozz) [inline src/Generator/AbstractGenerator.php]: `elseif` should be dettached from the `if` condition

File: src/Generator/Controller/Generator.php [hunk 2]
```diff
@@ -55,11 +56,14 @@ public function rules(): array
             [
                 'controllerClass' => [
                     new Required(),
-                    (new MatchRegularExpression('/^[\w\\\\]*Controller$/'))
-                        ->message('Only word characters and backslashes are allowed, and the class name must end with "Controller".')
+                    (new MatchRegularExpression('/^[\w]*Controller$/'))
```
Comment (xepozz) [inline src/Generator/Controller/Generator.php]: How about to check that the first letter is upper case?

Comment (rustamwin) [reply src/Generator/Controller/Generator.php]: done

**Comment by samdark** [comment]
👍

**Pull request merged**